code/datums/diseases/advance/symptoms/confusion.dm

```diff
@@ -55,7 +55,8 @@ Bonus
 			to_chat(M, "<span class='userdanger'>You can't think straight!</span>")
 			M.confused = min(100 * power, M.confused + 8)
 			if(brain_damage)
-				M.adjustBrainLoss(3 * power, 80)
+				var/obj/item/organ/brain = M.getorganslot(ORGAN_SLOT_BRAIN)
+				brain.applyOrganDamage(3 * power, 80)

```

Comment:
Null is a valid value for all organs, even brains.

Reply:
Additionally, you could keep adjustBrainLoss as a proc, but just make it do the above code.

Reply:
> 
> 
> getorganslot can return null for all organs, even brains.
> 
> Wouldn't it be neat if we had a proc that checked that kind of stuff when we apply damage to organs? Something like:
> 
> ```
> mob/living/carbon/proc/applyOrganDamage(slot, amount, maximum, required_status)
> 	var/item/organ/O = getorganslot(slot)
> 	if(O && (!required_status || O.status == required_status))
> 		O.applyOrganDamage(amount, maximum)
> ```
> 
> It was a mistake that such a proc was not added in the original organ damage PR.

Definitely agree, in application the applyOrganDamage shouldn't even be tied to the organ, not sure why I decided to do so. I'll move the proc from the organ to living and living/carbon

Reply:
I'm not adding required_status though, that seems far too niche to add when a coder can just check the status in an if statement. Adding it also means only a singular status is allowed, instead of CONSCIOUS/UNCONSCIOUS so I hope you don't mind.

Reply:
Kept the old procs too, since they work better when going through every organ for healing effects.

Reply:
Oh, status is refering to ORGANIC/INORGANIC, at least for organs.